### ISSUE TITLE
GitUp throwing 'npx not found' error on commit

### ISSUE THREAD

**Issue by gabcomby** [git-up/GitUp#1006]
The project I work on has a pre-commit hook which uses an npx command. When I use regular Git on my console, everything works fine. I use NVM, Noda 20, and running 'npx-v' correctly gives me the currently installed npx version, which shows it is indeed installed. However, when committing with GitUp, I get the following error: *npx: command not found*.

I tried reinstalling a fresh version of Node 20 with NVM, but to no avail. Any clues here?

**Comment by Cykelero** [comment]
It's likely that when it executes your hooks, GitUp doesn't have access to your environment variables, such as `PATH`; which in turn means it can't find the `npx` executable.

As a workaround (and a way to confirm this is indeed the cause!), you could try running the app through Terminal, which makes it inherit your shell's env variables:
```shell
$ nohup /Applications/GitUp.app/Contents/MacOS/GitUp > /dev/null 2>&1 &
```
(adjust the path depending on where you've installed the app)

**Comment by gabcomby** [comment]
@Cykelero when running the command, I get the following output, and my GitUp app doesn't launch:
[1] 52887
[1]  + exit 127   $ nohup /Applications/GitUp.app/Contents/MacOS/GitUp > /dev/null 2>&1

**Comment by Cykelero** [comment]
Sorry about the confusion: the `$` isn't part of the command, it's supposed to represent a shell prompt!
The actual command to copy and paste into Terminal starts at `nohup`, and it'll only work if GitUp is in the root Applications folder.

**Comment by gabcomby** [comment]
Yeah sorry that's on me, I wasn't really thinking lmao. It works great, now I'm able to commit freely. Is there any way to give GitUp access to my PATH? It would be rgeat if I could avoid having to launch it via Terminal every time...

**Comment by Cykelero** [comment]
I don't know of an easier workaround, sorry. GitUp would have to be updated with the ability to import your environment variables, but that's kind of thorny, as it can have side-effects.

(some resources if someone wants to work on that feature: Nova's [support page](https://help.panic.com/nova/environment-variables/) on the subject; [a SO answer](https://stackoverflow.com/questions/41535451/how-to-access-the-terminals-path-variable-from-within-my-mac-app-it-seems-to) with an example implementation; and [another related SO answer](https://stackoverflow.com/questions/54724324/access-user-environment-variable-in-swift-for-macos))

**Comment by fizker** [comment]
I have a crazy but functional workaround; I simply create a symlink in `/usr/local/bin` for `node` and `npm`. I have never needed `npx`, but just add that to the list and it should work as well.

I did this to make it work with running build scripts during Xcode compilation, so it should work for any app transparently without needing to update the app itself.

I execute this in `.zshrc` right after the `nvm` setup, so that it is mostly up-to-date with the latest version installed via `nvm`:

```zsh
# This exposes the nvm-installed node and npm to non-user programs when the shell is opened
# If nvm updates node/npm, it will be updated on the next new shell
if command -v node &> /dev/null && command -v nvm &> /dev/null
then
	mkdir -p ~/bin

	ln -fs "`which npm`" ~/bin/npm
	ln -fs "`which node`" ~/bin/node

	if [[ ! -a /usr/local/bin/node ]]; then
		echo 'Creating /usr/local/bin/node'
		sudo mkdir -p /usr/local/bin
		sudo ln -fs ~/bin/node /usr/local/bin/node
		sudo ln -fs ~/bin/npm /usr/local/bin/npm
	fi
fi
```
